clusterloader2/pkg/measurement/common/service_creation_latency.go

```diff
@@ -0,0 +1,413 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"fmt"
+	"sort"
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+
+	"k8s.io/perf-tests/clusterloader2/pkg/execservice"
+	"k8s.io/perf-tests/clusterloader2/pkg/framework"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	measurementutil "k8s.io/perf-tests/clusterloader2/pkg/measurement/util"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement/util/informer"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement/util/runtimeobjects"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement/util/workerqueue"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+const (
+	serviceCreationLatencyName           = "ServiceCreationLatency"
+	serviceCreationLatencyWorkers        = 10
+	defaultServiceCreationLatencyTimeout = 1 * time.Minute
+	defaultWaitTimeout                   = 10 * time.Minute
+	defaultCheckInterval                 = 10 * time.Second
+	pingBackoff                          = 1 * time.Second
+	pingChecks                           = 10
+)
+
+func init() {
+	measurement.Register(serviceCreationLatencyName, createServiceCreationLatencyMeasurement)
+}
+
+func createServiceCreationLatencyMeasurement() measurement.Measurement {
+	return &serviceCreationLatencyMeasurement{
+		queue:         workerqueue.NewWorkerQueue(serviceCreationLatencyWorkers),
+		creationTimes: make(map[string]*serviceCreationTimes),
+		pingCheckers:  make(map[string]*pingChecker),
+	}
+}
+
+type serviceCreationLatencyMeasurement struct {
+	namespace        string
+	labelSelector    string
+	fieldSelector    string
+	waitTimeout      time.Duration
+	stopCh           chan struct{}
+	isRunning        bool
+	selectorsString  string
+	queue            workerqueue.Interface
+	informer         cache.SharedInformer
+	clusterFramework *framework.Framework
+
+	lock          sync.Mutex
+	creationTimes map[string]*serviceCreationTimes
```

Comment:
I like it much more than those 5 maps:
https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/pkg/measurement/common/slos/pod_startup_latency.go#L66

if you could fix that in a separate PR, it would be great.

Reply:
https://github.com/kubernetes/perf-tests/pull/654 created.